src/tests/auth/login.test.js

```diff
@@ -0,0 +1,190 @@
+import chaiHttp from 'chai-http';
+import chai, { expect } from 'chai';
+import models from '../../models';
+import app from '../../app';
+import {
+  activeUser,
+  notActive,
+  wrongToken,
+  tokenWithWrongUser,
+  wrongUser
+} from '../fixtures/users.fixture';
+
+chai.use(chaiHttp);
+const router = () => chai.request(app);
+
+const { Users } = models;
+
+let user = null;
+let token = null;
+const userPassword = 'Pass1234';
+
+
+describe('Tests for user login', () => {
+  before(async () => {
+    await Users.create(activeUser);
+    await Users.create(notActive);
+    user = {
+      email: activeUser.email,
+      password: userPassword,
+    };
+  });
+  it('should login a user', (done) => {
+    router()
+      .post('/api/auth/login')
+      .send(user)
+      .end((err, res) => {
+        token = res.body.data;
+        expect(res).to.have.status(200);
+        expect(res.body).to.be.an('object');
+        expect(res.body).to.have.property('data');
+        expect(res.body).to.have.property('message')
+          .that.contain('Successfully logged in.. redirecting');
+        done(err);
+      });
+  });
+  it('Should give err if a user not active', (done) => {
+    router()
+      .post('/api/auth/login')
+      .send({
+        email: notActive.email,
+        password: userPassword,
+      })
+      .end((err, res) => {
+        expect(res).to.have.status(400);
+        expect(res.body).to.be.an('object');
+        expect(res.body).to.have.property('message')
+          .that.contain('You have not activated your account. Please activate your account first');
+        done(err);
+      });
+  });
+  it('Should check if not token provided', (done) => {
+    router()
+      .get('/api/protected')
+      .set('Authorization', '')
+      .send({
+        email: activeUser.email,
+        password: userPassword,
+      })
+      .end((err, res) => {
+        expect(res).to.have.status(401);
+        expect(res.body).to.have.property('message')
+          .that.contain('No Token supplied');
+        done(err);
+      });
+  });
+  it('Should not authorize wrong token', (done) => {
+    router()
+      .get('/api/protected')
+      .set('Authorization', wrongToken)
+      .send({
+        email: activeUser.email,
+        password: userPassword,
+      })
+      .end((err, res) => {
+        expect(res).to.have.status(401);
+        expect(res.body).to.have.property('message')
+          .that.contain('Unauthorized access. Invalid token');
+        done(err);
+      });
+  });
+  it('Should decode correct token', (done) => {
+    router()
+      .get('/api/protected')
+      .set('Authorization', token)
+      .send({
+        email: activeUser.email,
+        password: userPassword,
+      })
+      .end((err, res) => {
+        expect(res).to.have.status(200);
+        done(err);
+      });
+  });
+  it('Should slice \'Bearer\' from token token with it', (done) => {
```

Comment:
This test case title is not descriptive enough. How is this different from testing that a user is successfully logged in?

Reply:
it for testing this piece of code 
```
if (token.startsWith('Bearer ')) {
        // Remove Bearer from string
        token = token.slice(7, token.length);
      }
```
I am assuming that when user try to access protected route, we may get token from different places(eq.headers.authorization || req.headers['x-access-token'] || req.body.token || req.query.slt;)
some headers send the token with bearer before it then I needed to slice it in first place.

Reply:
This was solved in down comments. I don't need to slice anymore